Data/Level/LevelObject.cs

```diff
@@ -101,167 +101,123 @@ public enum RenderType
-        public byte bClamp;
-        public byte bPlasma;
-        public byte bSound;
-        public byte bRandom;
-        public byte bInPlane;
-        public Color color;
-        public bool enabled;
-    }
-
-    // D2X-XL
-    public struct SoundInfo
-    {
-        public string filename;
-        // Expected range 0-10, indicates how loud the sound should be
-        public int volume;
-        public bool enabled;
-    }
-
-    //Render info
-    public class PolymodelInfo
-    {
-        public int modelNum;
-        public FixAngles[] animAngles = new FixAngles[Polymodel.MAX_SUBMODELS];
-        public int flags;
-        public int textureOverride;
-    }
-
-    public struct SpriteInfo
-    {
-        public int vclipNum;
-        public Fix frameTime;
-        public byte frameNumber;
-    }
-
     public class LevelObject
     {
-        public ObjectType type;
-        public byte id;
+        /// <summary>
+        /// The object's basic type.
+        /// </summary>
+        public ObjectType Type { get; set; }
+        /// <summary>
+        /// The ID of the object's subtype. Meaning is based on the value of Type.
+        /// </summary>
+        public byte ID { get; set; }
 
-        public ControlType controlType;
-        public MovementType moveType;
-        public RenderType renderType;
-        public byte flags;
+        /// <summary>
+        /// Gets the numeric ID of the current ControlType.
+        /// </summary>
+        public ControlTypeID ControlTypeID
+        {
+            get
+            {
+                if (ControlType == null)
+                    return ControlTypeID.None;
+                return ControlType.Identifier;
+            }
+        }
+        /// <summary>
+        /// Gets the numeric ID of the current MoveType.
+        /// </summary>
+        public MovementTypeID MoveTypeID
+        {
+            get
+            {
+                if (RenderType == null)
+                    return MovementTypeID.None;
+                return MoveType.Identifier;
+            }
+        }
+        /// <summary>
+        /// Gets the numeric ID of the current RenderType.
+        /// </summary>
+        public RenderTypeID RenderTypeID
+        {
+            get
+            {
+                if (RenderType == null)
+                    return RenderTypeID.None;
+                return RenderType.Identifier;
+            }
+        }
+
+        /// <summary>
+        /// Misc object flags. Not generally useful in level files.
+        /// </summary>
+        public byte Flags { get; set; }
 
         /// <summary>
         /// Indicates if this object is only present in multiplayer modes. D2X-XL only.
         /// </summary>
         public bool MultiplayerOnly { get; set; }
-        public short segnum;
-        public short attachedObject;
+        /// <summary>
+        /// Number of the segment containing this object.
+        /// </summary>
+        public short Segnum { get; set; }
```

Comment:
I'm wondering about making this a pointer to a Segment. That would require deeper changes to Level.IO and probably some tests though.

Reply:
I suspect this would be best for a future refactoring? Not sure though. It's a good idea though, since it would bulletproof issues with segnum falling out of sync when segments are added and deleted in an editor.